API/hermes_abi/hermes_abi.h

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#ifndef HERMES_ABI_HERMES_ABI_H
+#define HERMES_ABI_HERMES_ABI_H
+
+#include <assert.h>
+#include <stdalign.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#ifndef HERMES_EXPORT
+#ifdef _MSC_VER
+#define HERMES_EXPORT __declspec(dllexport)
+#else // _MSC_VER
+#define HERMES_EXPORT __attribute__((visibility("default")))
+#endif // _MSC_VER
+#endif // !defined(HERMES_EXPORT)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Define the structure for references to pointer types in JS (e.g. string,
+/// object, BigInt).
+/// TODO: Replace jsi::PointerValue itself with this C implementation to
```

Comment:
In my experiments I saw that we can `reinterpret_cast` C++ `jsi::PointerValue` to a struct that has `HermesABIManagedPointerVTable` pointer as long as we specify explicit calling conventions.

Reply:
While it may work on certain platforms, we certainly shouldn't rely on it. I'm planning to replace `jsi::PointerValue` with `HermesABIManagedPointer` in the future, which should eliminate most of the wrapper overhead.